src/components/form/dt-users-connection/dt-users-connection.js

```diff
@@ -52,15 +63,22 @@ export class DtUsersConnection extends DtTags {
         }
         return result;
       }, null);
-      if (option) {
+
+      const singleValue = (this.value || [])
+      .filter(i => !i.delete);
+
+      if (option && (!this.single || singleValue.length < 1)) {
         this._select(option);
       }
```

Comment:
Based on the conversation in the issue, instead of ignoring the click of options in the list when it is single, the option clicked should replace the current value. So if single and already selected, de-select the previous value first and then you can do `this._select` like normal.